lib/rules/quote-props.js

```diff
@@ -123,7 +123,7 @@ module.exports = function(context) {
             var key = property.key,
                 tokens;
 
-            if (property.method || property.computed || property.shorthand) {
+            if (property.method || property.computed || property.shorthand || property.type === "ExperimentalSpreadProperty") {
```

Comment:
How about we simply add `key &&` to the front and cover some other possible scenarios?